x-pack/plugins/ml/public/jobs/new_job/simple/components/enable_model_plot_checkbox/enable_model_plot_checkbox_directive.js

```diff
@@ -0,0 +1,147 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import React from 'react';
+import ReactDOM from 'react-dom';
+
+import { EnableModelPlotCheckbox } from './enable_model_plot_checkbox_view.js';
+import { ml } from 'plugins/ml/services/ml_api_service';
+
+import { uiModules } from 'ui/modules';
+const module = uiModules.get('apps/ml');
+
+module.directive('mlEnableModelPlotCheckbox', function () {
+  return {
+    restrict: 'AE',
+    replace: false,
+    scope: {
+      formConfig: '=',
+      ui: '=ui',
+      getJobFromConfig: '='
+    },
+    link: function ($scope, $element) {
+      const STATUS = {
+        FAILED: -1,
+        NOT_RUNNING: 0,
+        RUNNING: 1,
+        FINISHED: 2,
+        WARNING: 3,
+      };
+      const errorHandler = (error) => {
+        console.log('Cardinality could not be validated', error);
+        $scope.ui.cardinalityValidator.status = STATUS.FAILED;
+        $scope.ui.cardinalityValidator.message = 'Cardinality could not be validated';
+      };
+
+      // Only model plot cardinality relevant
+      // format:[{id:"cardinality_model_plot_high",modelPlotCardinality:11405}, {id:"cardinality_partition_field",fieldName:"clientip"}]
+      const checkCardinalitySuccess = (data) => {
+        const response = {
+          success: true,
+        };
+        // There were no fields to run cardinality on.
+        if (Array.isArray(data) && data.length === 0) {
+          return response;
+        }
+
+        for (let i = 0; i < data.length; i++) {
+          if (data[i].id === 'success_cardinality') {
```

Comment:
It would be nice if these IDs used for job validation messages were in a constants file, but not needed for this PR.

Reply:
Sounds good for the follow up PR with the advanced wizard. 👍